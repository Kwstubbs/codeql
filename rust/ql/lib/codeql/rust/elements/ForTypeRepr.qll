// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
