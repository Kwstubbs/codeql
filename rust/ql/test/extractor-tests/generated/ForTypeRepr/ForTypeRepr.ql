// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  ForTypeRepr x, string hasGenericParamList__label, string hasGenericParamList,
  string hasTypeRepr__label, string hasTypeRepr
) {
  toBeTested(x) and
  not x.isUnknown() and
  hasGenericParamList__label = "hasGenericParamList:" and
  (if x.hasGenericParamList() then hasGenericParamList = "yes" else hasGenericParamList = "no") and
  hasTypeRepr__label = "hasTypeRepr:" and
  if x.hasTypeRepr() then hasTypeRepr = "yes" else hasTypeRepr = "no"
}

query predicate getGenericParamList(ForTypeRepr x, GenericParamList getGenericParamList) {
  toBeTested(x) and not x.isUnknown() and getGenericParamList = x.getGenericParamList()
}

query predicate getTypeRepr(ForTypeRepr x, TypeRepr getTypeRepr) {
  toBeTested(x) and not x.isUnknown() and getTypeRepr = x.getTypeRepr()
}
