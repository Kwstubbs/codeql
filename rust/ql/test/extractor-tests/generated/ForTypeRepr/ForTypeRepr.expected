instances
| gen_for_type_repr.rs:8:19:8:31 | ForTypeRepr |
getForBinder
| gen_for_type_repr.rs:8:19:8:31 | ForTypeRepr | gen_for_type_repr.rs:8:19:8:25 | for<...> |
getTypeRepr
| gen_for_type_repr.rs:8:19:8:31 | ForTypeRepr | gen_for_type_repr.rs:8:27:8:31 | usize |
